### PR TITLE
Implement game service and use it for tetris

### PR DESCRIPTION
**Pull request by fsmanuel**
Abstracts the join a game logic in a game service. If you like it I can also make the changes for `snake`.

Whenever there is no playerConnection all connections are able to start the game.

### CODE REVIEW

File: app/components/tetris/index.gts [hunk 1]
```diff
@@ -43,9 +44,17 @@ export class Tetris extends Component<TetrisSignature> {
     this.tetrominoSequence,
   );
 
+  constructor() {
```
Comment (kennstenicht) [inline app/components/tetris/index.gts:47]: Hier musst du vermutlich (owner: Tetris, args: TetrisSignature['Args']) definieren und dann  super(owner, args); verwneden.

Comment (fsmanuel) [reply app/components/tetris/index.gts:47]: Ich hab das von hier übernommen: https://docs.ember-cli-typescript.com/ts/using-ts-effectively#class-property-setup-errors

> Assertion Failed: Attempting to lookup an injected property on an object without a container, ensure that the object was instantiated via a container. – You failed to pass `...arguments` when you called `super` in e.g. a component class `constructor`. Always do `super(...arguments)`, not just `super()`, in your constructor.

Comment (fsmanuel) [reply app/components/tetris/index.gts:47]: Der Fehler ist:

> TS2556: A spread argument must either have a tuple type or be passed to a rest parameter.

Comment (fsmanuel) [reply app/components/tetris/index.gts]: Hab es jetzt wie hier beschrieben gemacht: https://docs.ember-cli-typescript.com/ember/components#adding-arguments-and-giving-them-a-type

**Pull request merged**